File: problems/383/solution.go
package main
func canConstruct(ransomNote string, magazine string) bool {
    m :=map[int]int{}
    
    for _, v := range magazine {
        m[int(v)]++
    }
    
    for _, v := range ransomNote {
        v1, ok := m[int(v)]
        
        if ok && v1 > 0 {
            m[int(v)]--
        } else {
            return false
        }
    }
    
    return true
}

File: problems/383/solution.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        d = defaultdict(int)
        
        for ch in magazine:
            d[ch] += 1
        
        for ch in ransomNote:
            if d[ch]:
                d[ch] -= 1
            else:
                return False
        
        return True